ksqldb-engine/src/main/java/io/confluent/ksql/physical/pull/HARouting.java

```diff
@@ -219,6 +221,16 @@ private void executeRounds(
           Preconditions.checkState(fetchResult.getResult() == RoutingResult.SUCCESS);
           processedPartitions++;
         }
+      } catch (TimeoutException timeoutException) {
+        final MaterializationException exception =
+            new MaterializationException(
+            "Query timeout exception");
```

Comment:
would it help to specify which node timed out so they can debug issues with connectivity etc...?